app/routes.py

```diff
@@ -1,2 +1,342 @@
+
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if not valid_item(customer) or not valid_item(video):
+        return make_response("Customer or video not found", 404)
+
+    new_rental = Rental(customer_id=customer_id, video_id=video_id)
+
+    if video.available_inventory == 0:
+        return {
+            "errors": ["No videos available"]
+        }, 400
+    else:
+        customer.videos_checked_out_count += 1
+        video.available_inventory -= 1
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    return {
+        "customer_id": customer.customer_id,
+        "video_id": video.video_id,
+        "due_date": new_rental.due_date,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+    }
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def check_in():
+    '''
+    INPUT: JSON with customer_id and video_id
+    OUTPUT: JSON with updated inventory information from customer and video
+
+    Decreases customer's videos_checked_out by 1
+    Increases video's available_inventory
+    '''
+    request_body = request.get_json()
+
+    if not valid_data_type(request_body["customer_id"], int) \
+    or not valid_data_type(request_body["video_id"], int):
+        return jsonify("IDs must be an integer"), 400
+
+    customer = Customer.query.get(request_body["customer_id"])
+    video = Video.query.get(request_body["video_id"])
+
+    if not valid_item(customer) or not valid_item(video):
+        return make_response("Customer or video not found", 400)
+
+    if customer.videos_checked_out_count == 0:
+        return jsonify("You don't have anything to return!"), 400
+    elif video.available_inventory >= video.total_inventory:
+        return jsonify("Did you return this to the wrong store?"), 400
+    else:
+        customer.videos_checked_out_count -= 1
+        video.available_inventory += 1
+
+    db.session.commit()
+
+    return {
+        "customer_id": customer.customer_id,
+        "video_id": video.video_id,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+    }
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def get_customer_videos(customer_id):
+    '''
+    Gets list of videos a specific customer has checked out
+    '''
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return make_response("", 400)
+
+    video_list = [] # list of videos currently rented by a particular customer
+
+    rental_results = db.session.query(Customer, Video, Rental).join(
+        Customer, Customer.customer_id==Rental.customer_id).join(
+        Video, Video.video_id==Rental.video_id).filter(
+        Customer.customer_id==customer_id).all()
```

Comment:
With a relationship this can be done more simply.

https://stackoverflow.com/questions/25375179/one-to-many-flask-sqlalchemy